metaflow/plugins/datatools/s3/s3.py

```diff
@@ -792,6 +792,59 @@ def _head():
 
         return list(starmap(S3Object, _head()))
 
+    def delete(self, key):
+        """
+        Delete a single object in S3.
+
+        Parameters
+        ----------
+        key : str
+            Object to delete. It can be an S3 url or a path suffix.
+        """
+
+        url = self._url(key)
+        src = urlparse(url)
+
+        def _delete(s3, tmp):
+            resp = s3.delete_object(Bucket=src.netloc, Key=src.path.lstrip('/'))
+
+            return resp
+
+        self._one_boto_op(_delete, url)
```

Comment:
we can probably set `create_tmp_file=False` here since we don't need any temporary file.